Gems/AtomLyIntegration/EditorModeFeedback/Assets/Shaders/EditorModeCommon.azsli

```diff
@@ -76,4 +145,23 @@ partial ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
     {
         return CalculateOutputColor(inColor, finalEffect, PassSrg::CalculateFinalBlendAmount(t));
     }
-}
\ No newline at end of file
+    
+    //! Calculates the blend amount (0.0 = input color, 1.0 = final effect) as scaled by the fragment's linear depth according to the depth transition values.
+    float CalculateTransitionBlendAmountFromLinearDepth(const float linearDepth, const float maskValue)
+    {
+        if(PassSrg::m_depthTransitionDuration > 0.0)
+        {
+            const float depthTransition = clamp((linearDepth - PassSrg::m_depthTransitionStart) / (PassSrg::m_depthTransitionDuration), PassSrg::m_minDepthTransitionValue, 1.0);
+            return clamp((1.0 - maskValue) * depthTransition, 0.0, 1.0);
+        }
+
+        return 1.0 - maskValue;
+    }
+
+    //! Calculates the blend amount (0.0 = input color, 1.0 = final effect) as scaled by the fragment's non-linear depth according to the depth transition values.
+    float CalculateTransitionBlendAmountFromDepth(const float zDepth, const float maskValue)
```

Comment:
Is the maskValue guaranteed to be between 0 and 1?

Reply:
It seems that way. The mask is provided by red channel of texture which is R8G8_unorm and there is no range modifications happening.